astroquery/esa/euclid/core.py

```diff
@@ -0,0 +1,1205 @@
+                                          tap_context="tap-server",
+                                          upload_context="Upload",
+                                          table_edit_context="TableTool",
+                                          data_context="cutout",
+                                          datalink_context="datalink",
+                                          verbose=verbose,
+                                          client_id='ASTROQUERY',
+                                          use_names_over_ids=conf.USE_NAMES_OVER_IDS)
+        else:
+            self.__euclidcutout = cutout_handler
+
+        if show_server_messages:
+            self.get_status_messages()
+
+    def launch_job(self, query, *, name=None, dump_to_file=False, output_file=None, output_format="csv", verbose=False,
+                   upload_resource=None, upload_table_name=None):
+        """
+        Description
+        -----------
+        Launches a synchronous job
+
+        Parameters
+        ----------
+        query : str, mandatory
+            query to be executed
+        name : str, optional, default None
+            custom name defined by the user for the job that is going to be created
+        dump_to_file : bool, optional, default 'False'
+            if True, the results are saved in a file instead of using memory
+        output_file : str, optional, default None
+            File name where the results are saved if dump_to_file is True.
+            If this parameter is not provided, the job id is used instead
+        output_format : str, optional, default 'csv'
+            output format for the output file
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+        upload_resource: str, optional, default None
+            resource to be uploaded to UPLOAD_SCHEMA
+        upload_table_name: str, required if uploadResource is provided, default None
+            resource temporary table name associated to the uploaded resource
+
+        Returns
+        -------
+        A Job object
+        """
+        try:
+            return super().launch_job(query=query, name=name,
+                                      output_file=output_file,
+                                      output_format=output_format,
+                                      verbose=verbose,
+                                      dump_to_file=dump_to_file,
+                                      upload_resource=upload_resource,
+                                      upload_table_name=upload_table_name,
+                                      format_with_results_compressed=('votable_gzip',))
+
+        except HTTPError as err:
+            log.error(f'Query failed: {query}: HTTP error: {err}')
+        except Exception as exx:
+            log.error(f'Query failed: {query}, {str(exx)}')
+
+    def launch_job_async(self, query, *, name=None, dump_to_file=False, output_file=None, output_format="csv",
```

Comment:
Not necessarily as part of this PR, but I wonder if we could/should merge the async and sync methods into one and have an `async_job` kwarg instead that switched between the two?